src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,82 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  setIsClicked: (isSelected: boolean) => void;
+  selectedTodo: Todo | null;
+  setSelectedUserId: (id: number) => void
+};
+
+export const TodoModal: React.FC<Props> = ({
+  setIsClicked,
+  selectedTodo,
+  setSelectedUserId: setSelectedId,
+}) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  useEffect(() => {
+    if (selectedTodo) {
+      getUser(selectedTodo.userId)
+        .then(user => {
+          setSelectedUser(user);
+          setIsLoaded(true);
+        });
+    }
+  }, []);
+
+  const isTodo = selectedTodo && selectedUser;
```

Comment:
it's not clear, what this variable should store, think about better variable name